python/tests/extras/test_image_metric.py

```diff
@@ -0,0 +1,90 @@
+
+import logging
+import os
+from typing import Dict
+
+logger = logging.getLogger(__name__)
+
+try:
+    from PIL.Image import Image as ImageType
+    from PIL.ImageStat import Stat
+    from PIL.TiffImagePlugin import IFDRational
+    from PIL.TiffTags import TAGS
+except ImportError as e:
+    ImageType = None
+    logger.debug(str(e))
+    logger.debug("Unable to load PIL; install Pillow for image support")
+
+
+import whylogs as why
+from whylogs.core.configs import SummaryConfig
+from whylogs.core.dataset_profile import DatasetProfile
+from whylogs.core.datatypes import DataType
+from whylogs.core.metrics import Metric, MetricConfig
+from whylogs.core.metrics.unicode_range import _STRING_LENGTH, UnicodeRangeMetric
+from whylogs.core.preprocessing import ListView, PreprocessedColumn
+from whylogs.core.resolvers import Resolver
+from whylogs.core.schema import ColumnSchema, DatasetSchema
+
+from whylogs.extras.image_metric import (
+    ImageMetric, 
+    get_pil_image_statistics,
+    get_pil_image_metadata,
+    image_based_metadata,
+    log_image,
+)
+
+
+
+TEST_DATA_PATH = os.path.abspath(
+    os.path.join(
+        os.path.realpath(os.path.dirname(__file__)),
+        os.pardir,
+        "testdata",
+    )
+)
+
+
+def image_loader(path: str = None) -> ImageType:
+    from PIL import Image  # to throw if PIL's not available
+
+    with open(path, "rb") as file_p:
+        img = Image.open(file_p).copy()
+        return img
+
+
+class TestResolver(Resolver):
+    def resolve(self, name: str, why_type: DataType, column_schema: ColumnSchema) -> Dict[str, Metric]:
+        return {ImageMetric.get_namespace(MetricConfig()): ImageMetric.zero(column_schema.cfg)}
+
+
+def test_image_metric() -> None:
+    image_path = os.path.join(TEST_DATA_PATH, "images", "flower2.jpg")
```

Comment:
I think we should remove the other images not used in the tests. We can add those images to the datasets module (stored in s3) when we introduce a notebook example.

Reply:
looks better, but you still have these two which I don't think we need:
python/tests/testdata/images/16bit.cropped.tif
python/tests/testdata/images/pal4.bmp